neo/rawio/blackrockrawio.py

```diff
@@ -565,25 +544,37 @@ def _parse_header(self):
                 # so datetime is valide only for seg_index=0
                 seg_ann['rec_datetime'] = rec_datetime
 
-            for c in range(sig_channels.size):
-                nsx_nb = sig_channels['group_id'][c]
-                anasig_an = seg_ann['signals'][c]
-                desc = "AnalogSignal {} from channel_id: {}, label: {}, nsx: {}".format(
-                    c, sig_channels['id'][c], sig_channels['name'][c], nsx_nb)
-                anasig_an['description'] = desc
-                anasig_an['file_origin'] = self._filenames['nsx'] + '.ns' + str(nsx_nb)
-                anasig_an['nsx'] = nsx_nb
-                chidx_ann = self.raw_annotations['signal_channels'][c]
-                chidx_ann['description'] = 'Container for Units and AnalogSignals of ' \
-                                           'one recording channel across segments.'
-
-            for c in range(unit_channels.size):
+            for c in range(signal_streams.size):
+                sig_ann = seg_ann['signals'][c]
+                stream_id = signal_streams['id'][c]
+                nsx_nb = int(stream_id)
+                sig_ann['description'] =  f'AnalogSignal from  nsx{nsx_nb}'
+                sig_ann['file_origin'] = self._filenames['nsx'] + '.ns' + str(nsx_nb)
+                sig_ann['nsx'] = nsx_nb
```

Comment:
Shouldn't the nsx info be also an array_annotation as in principle signal of multiple nsx files could be in a single stream if they share the same sampling rate etc?

Reply:
No here each nsx is a differents stream. It make sens.
So this nsx is not array annotation.